llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -495,10 +495,8 @@ class TargetTransformInfo {
 
   LLVM_ABI bool isNoopAddrSpaceCast(unsigned FromAS, unsigned ToAS) const;
 
-  std::optional<uint64_t> getAddrSpaceCastMaskWidth(unsigned FromAS,
-                                                    unsigned ToAS,
-                                                    Value *MaskOP,
-                                                    Instruction *I) const;
+  LLVM_ABI KnownBits computeKnownBitsAddrSpaceCast(unsigned FromAS,
```

Comment:
This should pass the source value, the target may be able to make a better determination based on it